sdk/purview/purview-administration-rest/test/public/utils/recordedClient.ts

```diff
@@ -33,52 +24,34 @@ const replaceableVariables: { [k: string]: string } = {
   AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
 };
 
-export const environmentSetup: RecorderEnvironmentSetup = {
-  replaceableVariables,
-  customizationsOnRecordings: [
-    (recording: string): string =>
-      recording.replace(/"access_token"\s?:\s?"[^"]*"/g, `"access_token":"access_token"`),
-    // If we put ENDPOINT in replaceableVariables above, it will not capture
-    // the endpoint string used with nock, which will be expanded to
-    // https://<endpoint>:443/ and therefore will not match, so we have to do
-    // this instead.
-    (recording: string): string => {
-      const replaced = recording.replace("endpoint:443", "endpoint");
-      return replaced;
-    },
-  ],
-  queryParametersToSkip: [],
+const recorderOptions: RecorderStartOptions = {
+  envSetupForPlayback: replaceableVariables,
 };
 
-export function createAccountClient(
+export async function createAccountClient(
+  recorder: Recorder,
   options?: ClientOptions
-): PurviewAccount.Client.PurviewAccountRestClient {
-  const credential = new ClientSecretCredential(
-    env.AZURE_TENANT_ID,
-    env.AZURE_CLIENT_ID,
-    env.AZURE_CLIENT_SECRET,
-    { httpClient }
+): Promise<PurviewAccount.Client.PurviewAccountRestClient> {
+  const credential = createTestCredential();
+  await recorder.start(recorderOptions);
+
+  return PurviewAccountClient(
+    env.ENDPOINT ?? "",
+    credential,
+    recorder.configureClientOptions({ httpClient, ...options })
```

Comment:
Since we are migrating to the new recorder, we can get rid of the `httpClient`. Please change this to `recorder.configureClientOptions(options)` and remove `const httpClient = isNode || isLiveMode() ? undefined : createXhrHttpClient();` above.

You'll have to re-record after this change

This was a workaround while these migrations happened because the old recorder didn't support Fetch API. Now the ner recorder does so we no longer need to force `xhrHttpClient`

Reply:
Got it! Thank you for the explanation :)